x/lockup/keeper/lock_test.go

```diff
@@ -10,12 +10,143 @@ import (
 	"github.com/osmosis-labs/osmosis/osmomath"
 	cl "github.com/osmosis-labs/osmosis/v19/x/concentrated-liquidity"
 	cltypes "github.com/osmosis-labs/osmosis/v19/x/concentrated-liquidity/types"
+	"github.com/osmosis-labs/osmosis/v19/x/lockup/keeper"
 	"github.com/osmosis-labs/osmosis/v19/x/lockup/types"
 	lockuptypes "github.com/osmosis-labs/osmosis/v19/x/lockup/types"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+func (suite *KeeperTestSuite) TestRebondTokens() {
```

Comment:
Needs test case for lock that is not owned by the caller

Also will need a test case to show the rewardAddr is static after the rebond, regaurdless of partial or full (so the test should have the rewardAddr be diff than the owner to prove this)

Reply:
added test case for non-owner message sender, added assertion that rewardReceiver address does not change after rebonding